packages/content-management/favorites/favorites_server/src/favorites_usage_collection.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CoreSetup } from '@kbn/core-lifecycle-server';
+import type { Logger } from '@kbn/logging';
+import type { UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+import type { estypes } from '@elastic/elasticsearch';
+import { favoritesSavedObjectName } from './favorites_saved_object';
+
+interface FavoritesUsage {
+  [favorite_object_type: string]: {
+    total: number;
+    total_users_spaces: number;
+    avg_per_user_per_space: number;
+    max_per_user_per_space: number;
+  };
+}
+
+export function registerFavoritesUsageCollection({
+  core,
+  logger,
+  usageCollection,
+}: {
+  core: CoreSetup;
+  logger: Logger;
+  usageCollection: UsageCollectionSetup;
+}) {
+  usageCollection.registerCollector(
+    usageCollection.makeUsageCollector<FavoritesUsage>({
+      type: 'favorites',
+      isReady: () => true,
+      schema: {
+        DYNAMIC_KEY /* e.g. 'dashboard' */: {
+          total: {
+            type: 'long',
+            _meta: { description: 'Total favorite object count in this deployment' },
+          },
+          total_users_spaces: {
+            type: 'long',
+            _meta: {
+              description:
+                'Total unique users+spaces that have favorited an object of this type in this deployment',
```

Comment:
I don't follow what `users+spaces` means. Does it mean users per space?

Reply:
I'll adjust!